app/src/main/java/com/example/sopt_assignment_dabin/ui/SignInActivity.kt

```diff
@@ -0,0 +1,209 @@
+package com.example.sopt_assignment_dabin.ui
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.animation.AnimationUtils
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.example.sopt_assignment_dabin.R
+import com.example.sopt_assignment_dabin.data.local.SigninRequestData
+import com.example.sopt_assignment_dabin.databinding.ActivitySignInBinding
+import com.example.sopt_assignment_dabin.sopt.SignServiceCreator
+import com.example.sopt_assignment_dabin.util.MyUtil.getHelper
+import com.example.sopt_assignment_dabin.util.MyUtil.shortToast
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var getResultText: ActivityResultLauncher<Intent>//회원가입 데이터 리턴받을 때 사용
+    private lateinit var binding: ActivitySignInBinding //해당 class에서만 사용되니 private으로 선언하는 습관
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setAnimation(this)
+
+        //모든 데이터가 입력되었을때 로그인 버튼 색깔 바뀌게하기->addTextChangedListener 사용
+        //인터페이스인 TextWatcher의 추상메서드인 3개함수 모두 오버라이드
+        binding.etPassIn.addTextChangedListener(object : TextWatcher {
+            @SuppressLint("ResourceAsColor")
+            override fun afterTextChanged(inputChar: Editable?) {
+            }
+
+            override fun beforeTextChanged(inputChar: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(inputChar: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                if (binding.etPassIn.text.isNotBlank() && binding.etIdIn.text.isNotBlank()) {
+                    binding.bvLogin.setBackgroundResource(R.drawable.rectangle_sopt_radius_8)
+                }
+            }
+        })
```

Comment:
```suggestion
        binding.etPassIn.doAfterTextChanged {
            if (binding.etPassIn.text.isNotBlank() && binding.etIdIn.text.isNotBlank()) {
                    binding.bvLogin.setBackgroundResource(R.drawable.rectangle_sopt_radius_8)
            }
        }
```

이렇게 확장함수의 형태로 만들 수 있습니다!

Reply:
대박적 너무 유용하네요 바로 반영했어요 감사합니다!